### PR TITLE
Documentation on writing unit tests

### PR DESCRIPTION
**Pull request by LockTar**
Fixes https://github.com/microsoft/kiota/issues/2767

As discussed with @baywet I created a first version on writing unit tests with a Kiota client. 
See the following issue: [https://github.com/microsoft/kiota/issues/2767](https://github.com/microsoft/kiota/issues/2767).

Original code can be found [here](https://github.com/LockTar/Microsoft-Graph-With-MSI-Sample/blob/main/Tests/MicrosoftGraphUserTests.cs).

Sample contains tests for:
- Get a list
- Get a single item
- Get a list of filtered items
- Post new item

### CODE REVIEW

File: OpenAPI/kiota/testing.md
Comment (baywet) [inline OpenAPI/kiota/testing.md:12]:
```suggestion
Code where Kiota API clients are used can be tested. In this document is explained how to mock the Microsoft Graph SDK.
```

File: OpenAPI/kiota/testing.md
Comment (baywet) [inline OpenAPI/kiota/testing.md:18]: @jasonjoh I wonder whether we should use one of the get started samples instead?

**Review by baywet — CHANGES_REQUESTED** [review]
Thank you a lot for this great contribution!
A couple of minor comments. @jasonjoh for final review

File: OpenAPI/kiota/testing.md
Comment (LockTar) [reply OpenAPI/kiota/testing.md:18]: You could also do a link to the [overview](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/dev/docs/overview.md) page?

Comment (baywet) [reply OpenAPI/kiota/testing.md:18]: I was referring to this description so it's consistent with the rest of the documentation https://learn.microsoft.com/en-us/openapi/kiota/quickstarts/dotnet#generate-the-api-client

Comment (LockTar) [reply OpenAPI/kiota/testing.md:18]: Ah ok. Yes that is something you as a team should decide  👍🏻

Comment (baywet) [reply OpenAPI/kiota/testing.md:18]: yep, Jason is our technical writer. However it's a holiday in the US today, expect some latency :)

Comment (jasonjoh) [reply OpenAPI/kiota/testing.md:18]: Yes, I think it makes sense to incorporate this into the quick start project and consume the code directly from there. I'll take a look at it.

Comment (LockTar) [reply OpenAPI/kiota/testing.md:18]: @jasonjoh added tests based on this PR in the sample project. See [https://github.com/microsoft/kiota-samples/pull/2854](https://github.com/microsoft/kiota-samples/pull/2854). Will resolve conversation.

Comment (baywet) [reply OpenAPI/kiota/testing.md:18]: Thanks for putting the pull request together Jason. Keeping this open as we need to update the snippets here once the pr is merged.

Comment (jasonjoh) [reply OpenAPI/kiota/testing.md]: Snippet replaced with a snippet reference, it will load the code from the sample project.

**Review by jasonjoh — APPROVED** [review]

**Pull request merged**
